### PR TITLE
Update market selector to produce UI properties.

### PR DESCRIPTION
**Pull request by bthaile**
https://github.com/AugurProject/augur/issues/477
https://github.com/AugurProject/augur/issues/2518

Change up marketsData reducer to take keyed object of MarketInfo objects. Market selector converts MarketInfo objects to MarketData UI objects.

Started using new Formatted UI properties. Fixed issues where UI was overriding properties from getter like `maxPrice`, `minPrice`, `volume`, `openInterest`, ....

created UI specific formatted properties like
`maxPriceBigNumber`, `minPriceBigNumber`, `volumeFormatted`, `openInterestFormatted`

fixed issues with outcome list selector

Added market loading modal, temporary until we get design.

### CODE REVIEW

**Comment by adrake33** [comment]
I updated Markets.test.ts & made some additional changes to Markets.ts. The changes related to the new properties in getMarketsInfo should be ready to go now.

**Review by adrake33 — APPROVED** [review]

File: packages/augur-ui/src/modules/market/components/market-view/market-view.tsx [hunk 1]
```diff
@@ -115,22 +118,14 @@ export default class MarketView extends Component<MarketViewProps, MarketViewSta
 
   componentDidMount() {
     this.node.scrollIntoView();
-
```
Comment (andrewdamelio) [inline packages/augur-ui/src/modules/market/components/market-view/market-view.tsx:118]: 👍

File: packages/augur-ui/src/modules/market/containers/market-view.ts [hunk 2]
```diff
@@ -1,29 +1,32 @@
-import { connect } from "react-redux";
-import { withRouter } from "react-router-dom";
-import MarketView from "modules/market/components/market-view/market-view";
-import { loadFullMarket } from "modules/markets/actions/load-full-market";
-import { selectMarket } from "modules/markets/selectors/market";
-import parseQuery from "modules/routes/helpers/parse-query";
-import { MARKET_ID_PARAM_NAME } from "modules/routes/constants/param-names";
+import { connect } from 'react-redux';
```
Comment (andrewdamelio) [inline packages/augur-ui/src/modules/market/containers/market-view.ts:1]: ```'react-redux';``` single quotes

Comment (bthaile) [reply packages/augur-ui/src/modules/market/containers/market-view.ts:1]: there are a much of single quote import stmt. we'll get that worked out when we get repo wide linting rules set.

**Pull request merged**